File: applications/luci-statistics/luasrc/model/cbi/luci_statistics/iwinfo.lua
--[[

Luci configuration model for statistics - collectd interface plugin configuration
(c) 2011 Jo-Philipp Wich <xm@subsignal.org>

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

        http://www.apache.org/licenses/LICENSE-2.0

$Id$

]]--

local m, s, o

m = Map("luci_statistics",
	translate("Wireless iwinfo Plugin Configuration"),
	translate("The iwinfo plugin collects statistics about wireless signal strength, noise and quality."))

s = m:section(NamedSection, "collectd_iwinfo", "luci_statistics")

o = s:option(Flag, "enable", translate("Enable this plugin"))
o.default = 0

o = s:option(Value, "Interfaces", translate("Monitor interfaces"),
	translate("Leave unselected to automatically determine interfaces to monitor."))
o.template = "cbi/network_ifacelist"
o.widget   = "checkbox"
o.nocreate = true
o:depends("enable", 1)

o = s:option(Flag, "IgnoreSelected", translate("Monitor all except specified"))
o.default = 0
o:depends("enable", 1)

return m

File: applications/luci-app-statistics/htdocs/luci-static/resources/view/statistics/plugins/iwinfo.js
'use strict';
'require baseclass';
'require form';
'require tools.widgets as widgets';

return baseclass.extend({
	title: _('Wireless iwinfo Plugin Configuration'),
	description: _('The iwinfo plugin collects statistics about wireless signal strength, noise and quality.'),

	addFormOptions: function(s) {
		var o;

		o = s.option(form.Flag, 'enable', _('Enable this plugin'));

		o = s.option(widgets.DeviceSelect, 'Interfaces', _('Monitor interfaces'), _('Leave unselected to automatically determine interfaces to monitor.'));
		o.multiple = true;
		o.noaliases = true;
		o.noinactive = true;
		o.depends('enable', '1');
		o.filter = function(section_id, name) {
			var dev = this.devices.filter(function(dev) { return dev.getName() == name })[0];
			return (dev && dev.getType() == 'wifi');
		};

		o = s.option(form.Flag, 'IgnoreSelected', _('Monitor all except specified'));
		o.depends('enable', '1');
	},

	configSummary: function(section) {
		var ifaces = L.toArray(section.Interfaces),
		    invert = section.IgnoreSelected == '1';

		if (ifaces.length == 0)
			return _('Monitoring all interfaces');
		else if (invert)
			return N_(ifaces.length, 'Monitoring all but one interface', 'Monitoring all but %d interfaces').format(ifaces.length);
		else
			return N_(ifaces.length, 'Monitoring one interface', 'Monitoring %d interfaces').format(ifaces.length);
	}
});
